src/pulselistener/pulselistener/lib/bus.py

```diff
@@ -59,6 +59,12 @@ def is_full(self):
         if isinstance(queue, asyncio.Queue):
             return await queue.get()
         else:
+
+            # This sleep call is needed to avoid fully blocking
+            # async process waiting on a multiprocessing queue
+            # Exemple: code review workflow waiting for new builds from webserver
+            await asyncio.sleep(0)
+
             return queue.get()
```

Comment:
That's what I was afraid of, but I think it would be better to use run_in_executor, so we don't block other asyncio tasks while we are waiting on the queue: https://github.com/mozilla/release-services/pull/2223#discussion_r308255234.

With the simple sleep, we allow other tasks to run before `queue.get()`, but we still block when calling `queue.get()`.

Reply:
https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor

Reply:
Alright, i'll try that in this PR. I *think* i made some unconclusive tests in the initial PR. But the smaller scope here should help.

Reply:
I think you just need something like:
```
return await asyncio.get_running_loop().run_in_executor(None, lambda: queue.get())
```